roles/keystone/tasks/main.yml

```diff
@@ -31,9 +31,49 @@
     - etype: other
       permission: r
 
-- name: install keystone service
-  upstart_service: name=keystone user=keystone cmd=/usr/local/bin/keystone-all
-                   config_dirs=/etc/keystone
+- name: install keystone uwsgi service
+  template: src=roles/keystone/templates/etc/init/keystone.conf
+            dest=/etc/init/keystone.conf mode=0644
+  with_items:
+    - { uwsgi_path: "{{ openstack_source.virtualenv_base }}/keystone/bin/uwsgi" }
+  when: openstack_install_method == 'source'
+  
+- name: install keystone uwsgi service
+  template: src=roles/keystone/templates/etc/init/keystone.conf
+            dest=/etc/init/keystone.conf mode=0644
+  with_items:
+    - { uwsgi_path: "{{ openstack_package.virtualenv_base }}/keystone/bin/uwsgi" }
+  when: openstack_install_method == 'package'
+
+- name: Creates keystone uwsgi directory
+  file: path=/etc/keystone/uwsgi state=directory
+        owner=keystone group=keystone mode=0775
+
+- name: Creates keystone httpd uwsgi directory
+  file: path=/var/www/keystone state=directory
+        owner=keystone group=keystone mode=0775
+
+- name: copy keystone httpd file to admin location
+  command: cp /opt/stack/keystone/httpd/keystone.py /var/www/keystone/admin
```

Comment:
This source file may not be in the package, and it's very small, we should probably just make it a template we house in ursula itself.